[WIP] GRIP engine embedded in a python library

kellrott opened the pull request:
TODO:
 - [x] setup.py
 - [x] Unit testing
 - [ ] pip install
 - [ ] Documentation
 - [ ] Better type/error checking
 - [ ] Fix error logging

kellrott committed:
Adding mem based key-value driver and testing a python lib

kellrott committed:
Working python wrapper library

kellrott committed:
Fixing client

kellrott committed:
Adding setup.py for pygrip

kellrott committed:
Working on adding unit tests

kellrott committed:
Doing user install for python

kellrott committed:
add step to actually run the pygrip unit tests

kellrott committed:
Adding missing file

kellrott committed:
Debugging unit test running

kellrott committed:
Removing stdout error statements

bwalsh commented on .github/workflows/tests.yml:
```diff
@@ -233,10 +233,28 @@ jobs:
+          python setup.py install --user
+      - name: install pygrip
+        run: |
+          python setup.py install --user
```
`python setup.py install --user` produces:

```
...
Installed /Users/walsbr/.local/lib/python3.9/site-packages/pygrip-0.8.0-py3.9-macosx-12.4-arm64.egg
Processing dependencies for pygrip==0.8.0
Searching for pygrip==0.8.0
Reading https://pypi.org/simple/pygrip/
No local packages or working download links found for pygrip==0.8.0
error: Could not find suitable distribution for Requirement.parse('pygrip==0.8.0')```

bwalsh replied:
Changed to 3.12, works now
```
Using /Users/walsbr/bmeg/grip/venv/lib/python3.12/site-packages
Finished processing dependencies for pygrip==0.8.0```

bwalsh replied:
However,  it doesn't look like it installed?

```
$ pip freeze | grep pygrip
>>>  (no hits)
```

``` 
python -c "import pygrip"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/walsbr/bmeg/grip/pygrip/__init__.py", line 9, in <module>
    from gripql.query import QueryBuilder
ModuleNotFoundError: No module named 'gripql.query'```

```
$ pytest test/pygrip_test/test_pygrip.py
========================================================================================== test session starts ==========================================================================================
platform darwin -- Python 3.12.1, pytest-8.1.1, pluggy-1.4.0
rootdir: /Users/walsbr/bmeg/grip
collected 0 items / 1 error

================================================================================================ ERRORS =================================================================================================
___________________________________________________________________________ ERROR collecting test/pygrip_test/test_pygrip.py ____________________________________________________________________________
ImportError while importing test module '/Users/walsbr/bmeg/grip/test/pygrip_test/test_pygrip.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../.pyenv/versions/3.12.1/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test/pygrip_test/test_pygrip.py:2: in <module>
    import pygrip
E   ModuleNotFoundError: No module named 'pygrip'
======================================================================================== short test summary info ========================================================================================
ERROR test/pygrip_test/test_pygrip.py```

bwalsh commented on .github/workflows/tests.yml:
```diff
@@ -233,10 +233,28 @@ jobs:
+      - name: Check out code
+        uses: actions/checkout@v2
+      - name: Python Dependencies for Conformance
+        run: pip install requests numpy PyYAML
```
May need to include `setuptools`

bwalsh reviewed (commented):
See questions.  I tried following steps in `pygripTest` setup but I think I'm missing a step

bwalsh committed:
adds pythonpath for pytest

bwalsh committed:
make ./test a package

bwalsh committed:
test FHIR load, dataframe traversals

test FHIR load

adds jsonpath-ng for reference searches

improves traversal tests

bwalsh committed:
adds comments and TODO

matthewpeterkort commented:
Tests pass. Seems like the data frame association function lists for every vertex all of the edges that are 1 edge away from each other. 

In order to make this strategy more flexible like the data-framer is currently, the ```dataframe_associations``` function would need to be significantly built out to detect fhir edge locations in the dict and aggregate associations automatically. 

Can see how adding edge search via FHIR schema Rel paths would aide in this process.

kellrott committed:
Merge branch 'develop' into feature/python-lib

kellrott committed:
Adding python dependencies to unit tests

kellrott committed:
Merge branch 'develop' into feature/python-lib

kellrott committed:
Eliminating debugging Printf statements

kellrott committed:
Trying to put togeather a grpc based embedded grip
